pkg/webhook/resources/upgrade/validator.go

```diff
@@ -224,6 +238,37 @@ func (v *upgradeValidator) checkManagedCharts() error {
 	return nil
 }
 
+// Since volume snapshot/backup will hold one additional LH VA tickets, it can block VM live migration.
+// We should check if there is any vmbackup under processing before upgrade
+func (v *upgradeValidator) checkVMBackups() error {
+	vmBackups, err := v.vmBackupCache.GetByIndex(indexeres.VMBackupByIsProgressing, strconv.FormatBool(true))
+	if err != nil {
+		return err
+	}
+
+	if len(vmBackups) == 0 {
+		return nil
+	}
+
+	return werror.NewBadRequest(fmt.Sprintf("vmbackup %s/%s is under processing", vmBackups[0].Namespace, vmBackups[0].Name))
```

Comment:
Is there any reason we only show the first vmbackup to the user?

Reply:
Because if the message contains all the running VMBackups, its length could be very long, and make it verbose.

In my opinion, showing the first vmbackup should provide enough information for the user to stop all running VMBackups. Even the user doesn't stop all the VMBackup, the next running VMBackup will be popped up by the webhook message

Reply:
Thank you. It makes sense to me. But I'm thinking maybe we can polish the error message a bit and give the user a hint that there might be more than this one to handle. So the user doesn't see similar errors again and then realizes they need to stop **all** the VMBackups/ScheduleVMBackups.

Reply:
Hi @starbops,
I updated the webhook message, please take a look, thanks.